wg-api/spec-documents/scroll-view.md

```diff
@@ -0,0 +1,178 @@
+* `removeChildView(view)`
+* `getViews()`
+
+- - -
+
+### ContainerView
+
+A `ContainerView` can be used to embed additional views hierarchy into a `BrowserWindow`. It extends `BaseView`
+
+Process: **Main**
+
+#### Static Methods
+
+* `ContainerView.fromId(id)` 
+
+#### Instance Methods
+
+* `addChildView(view)`
+* `removeChildView(view)`
+* `getViews()` - Returns an array of all BaseViews that have been attached
+
+
+### ScrollView
+
+Show a part of view with scrollbar.  The `ScrollView` can show an arbitrary content view inside it. It is used to make any View scrollable. When the content is larger than the `ScrollView`, scrollbars will be optionally showed. When the content view is smaller then the `ScrollView`, the content view will be resized to the size of the `ScrollView`. The scrollview supports keyboard UI and mousewheel. It extends `BaseView`.
+
+Process: **Main**
+
+
+#### Static Methods
+
+
+* `ScrollView.fromId(id)`
+
+#### Instance Methods
+
+* `setContentView(contents)` Set the contents. The contents is the view that needs to scroll.
+* `getContentView()`  * Returns The contents of the `view`.
+* `setHorizontalScrollBarMode(mode)` - possible values: `disabled`,  `hidden-but-enabled`, `enabled`. Default is `enabled` 
+  * `disable` - The scrollbar is hidden, and the pane will not respond to e.g. mousewheel events even if the contents are larger than the viewport.
+  * `hidden-but-enabled` - The scrollbar is hidden whether or not the contents are larger than the viewport, but the pane will respond to scroll events.
+  * `enabled` - The scrollbar will be visible if the contents are larger than the viewport and the pane will respond to scroll events.
+* `setVerticalScrollBarMode(mode)` - similar as `setHorizontalScrollBarMode`
+* `getHorizontalScrollBarMode()`
+* `getVerticalScrollBarMode()`
+* `setHorizontalScrollElasticity(elasticity)` (*macOS*) - possible values: `automatic`, `none`, `allowed`. Default is `automatic`.
+  
+  The scroll view’s horizontal scrolling elasticity mode.
+  A scroll view can scroll its contents past its bounds to achieve an elastic effect. 
+  When set to `automatic`, scrolling the horizontal axis beyond its document bounds only occurs if the document width is greater than the view width, or the vertical scroller is hidden and the horizontal scroller is visible.
+  * `automatic` - Automatically determine whether to allow elasticity on this axis.
+  * `none` - Disallow scrolling beyond document bounds on this axis.
+  * `allowed` - Allow content to be scrolled past its bounds on this axis in an elastic fashion.
+
+* `setVerticalScrollElasticity(elasticity)` (_macOS_) - similar to `setHorizontalScrollElasticity`
+* `getHorizontalScrollElasticity()` (_macOS_)
+* `getVerticalScrollElasticity()` (_macOS_)
+* `scrollTo({ x, y })`
+
+
+#### Events
+
+* `scroll-started`
+* `scroll-ended`
+
+
+### WrapperBrowserView
+
+A `WrapperBrowserView` is the wrapper for `BrowserView`. It extends [`BaseView`](base-view.md).
```

Comment:
Why is this wrapper necessary? It seems a bit awkward. Can we just use the BrowserView directly?

Reply:
I added WrapperBrowserView for clarity in the implementation. The thing is, on Mac, I defined the ElectronNativeView interface (inheriting from NSView), on the other hand, InspectableWebContentsViewMac creates ElectronInspectableWebContentsView.
I could get rid of WrapperBrowserView and add the addBrowserView / removeBrowserView functions to ContainerView instead.

Reply:
> I could get rid of WrapperBrowserView and add the addBrowserView / removeBrowserView functions to ContainerView instead.

I haven't thought through the implications so feel free to shoot this down, but could `addView()` just take `BaseView | BrowserView` instead and let the implementation call the right function  under the covers? If we have a new function named `addView()` then surely people are going to try passing BrowserViews into it

Reply:
What about `containerView.getChildren`? Should it return only BaseViews?
Maybe a better idea would be to replace WrapperBrowserView by WebView - inheriting from BaseView - with an interface analogous to BrowserView?